### PR TITLE
getTable() is static

### PR DESCRIPTION
**Pull request by btry**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | none

a static method should be called statically, even if PHP does not complains doing call other ways.

### CODE REVIEW

**Comment by btry** [comment]
lets replace $this::getTable() by static::getTable() first

File: inc/apiclient.class.php [hunk 1]
```diff
@@ -76,15 +76,15 @@ function getSearchOptionsNew() {
 
       $tab[] = [
          'id'                 => '1',
-         'table'              => $this->getTable(),
+         'table'              => $static::getTable(),
```
Comment (trasher) [inline inc/apiclient.class.php]: Hum... `$static` ;)

Comment (btry) [inline inc/apiclient.class.php]: indeed ! grr...

**Comment by remicollet** [comment]
Sorry, but I don't see any huge benefit for such change.

Ok to call is statically to avoid an object instantiation

```
- $this->getTable();
+ static::getTable();
```
The late static binding have a cost, uneeded when we have a object instance.

**Comment by remicollet** [comment]
From a quick bench, using $this is 5% faster

**Comment by trasher** [comment]
Thanks @remicollet! This is interesting; I did not have perfs in mind... 

On the whole app, 10% is quite important; so we can let those static calls as they are...

Finally we should maybe do the exact inverse; but there are many other tasks that can be accomplished to enhance performances :)

**Pull request closed without merging**